src/widgets/keyboard/lv_keyboard.c

```diff
@@ -287,58 +454,78 @@ void lv_keyboard_def_event_cb(lv_event_t * e)
         keyboard->mode = LV_KEYBOARD_MODE_TEXT_LOWER;
         lv_buttonmatrix_set_map(obj, kb_map[LV_KEYBOARD_MODE_TEXT_LOWER]);
         lv_keyboard_update_ctrl_map(obj);
-        return;
+        goto update_candidate;
```

Comment:
Instead of `goto` let's have a function for this large `if-else if` from which we can clearly return

Reply:
> Instead of `goto` let's have a function for this large `if-else if` from which we can clearly return

okay, i will try it